### ISSUE TITLE
401 Authorization Required When Accessing Models

### ISSUE THREAD

**Issue by tetter36** [strongloop/loopback-connector-postgresql#147]
I'm running into the same issue that can be found here:
http://stackoverflow.com/questions/36082372/loopback-authentication-error

    {
      "error": {
        "name": "Error",
        "status": 401,
        "message": "Authorization Required",
        "statusCode": 401,
        "code": "AUTHORIZATION_REQUIRED",
        "stack": "Error: Authorization Required\n    at 
      }
    }

I have an identical ACL and I have an identical DEBUG output. For some reason, the isAuthenticated() is returning false even when an Access Token has been provided in the request. If I remove the DENY ACL, everything works fine (as is expected). I'm assuming this is an issue with the connectors but I could be wrong.

Here are the versions I'm using:
    "loopback": "^2.29.0",
    "loopback-boot": "^2.6.5",
    "loopback-component-explorer": "^2.1.0",
    "loopback-component-storage": "^1.8.1",
    "loopback-connector-postgresql": "^2.4.1",
    "loopback-datasource-juggler": "^2.46.1",

**Comment by tetter36** [comment]
I'm not sure what was going on with my build but I cloned the code from the ACL example and started manipulating files one-by-one until I moved all of my files from my project into the example project and I was able to authenticate and query the data fine. I then moved all of the code back to my project and it is still working. I have no idea what could have caused the issue. Closing as I cannot replicate the issue now.

**Comment by tetter36** [comment]
I was finally able to replicate it. I pulled the gun a little too fast on my last comment. For some reason, if I have host set to: 127.0.0.1 or localhost it will not work and I get the 401 error. If I set the host as 0.0.0.0 the data is returned as expected. I have no idea why this would be the case.

**Comment by tetter36** [comment]
I finally reproduced this issue. I deleted my database and ran into the same issue again. This time, the weird issue with the URL wasn't appearing. I noticed that I can't correctly authenticate if the User Id I'm trying to authenticate with is 0. I didn't see this anywhere in the documentation, and seeing how the default Id when setting everything up is 0 this could cause issues for others. I'd recommend adding this to the documentation when creating a new user or changing the default to 1 instead of 0.

I now realize this is more of a Loopback issue and not a Loopback-Postgres issue. I'll be reposting in Loopback. Closing this issue.
